### PR TITLE
HTML-841: Errors with Form Namespace and Path

### PR DESCRIPTION
**Pull request by mogoodrich**
## Issue I worked on

see https://issues.openmrs.org/browse/HTML-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java [hunk 1]
```diff
@@ -86,32 +92,7 @@ public Drug getAnswerDrug() {
 	public void setAnswerDrug(Drug answerDrug) {
 		this.answerDrug = answerDrug;
 	}
-	
-	@Deprecated
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java:90]: Just removed this as it was deprecated and not used anywhere as far as I could tell

File: api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java [hunk 2]
```diff
@@ -121,6 +102,13 @@ public static int supportingRank(List<ObsGroupComponent> obsGroupComponents, Set
 			
 			// iterate though all form obs elements for obs group we are testing for a match against
 			for (ObsGroupComponent obsGroupComponent : obsGroupComponents) {
+				
+				// if any one of the obs group components have a control id, and it matches the obs control id, just return 1000 to force a match
+				if (obsGroupComponent.getControlId() != null
+				        && obsGroupComponent.getControlId().equals(HtmlFormEntryUtil.getControlId(obs))) {
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java:108]: Defer to you as to whether this `1000` value is sufficiently safe, vs. (eg) max value integer

**Review by mseaton — APPROVED** [review]
Overall looks good, though generally defer to your analysis

File: api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java (hunk 2 above)
Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/htmlformentry/ObsGroupComponent.java:108]: We used -1000 for the max negative, so just using this for consistency.  I'm pretty sure it sufficiently safe.

**Pull request merged**